pkg/apiserver/index.go

```diff
@@ -25,8 +25,17 @@ import (
 	"github.com/emicklei/go-restful"
 )
 
+const (
+	dashboardAlias = "/ui"
+	dashboardPath  = "/static/app/#/dashboard/"
+)
+
 func IndexHandler(container *restful.Container, muxHelper *MuxHelper) func(http.ResponseWriter, *http.Request) {
 	return func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path == "/index.html" || r.URL.Path == dashboardAlias || r.URL.Path == dashboardAlias+"/" {
+			http.Redirect(w, r, dashboardPath, http.StatusTemporaryRedirect)
```

Comment:
Sorry to mislead you. This redirect should be installed in the ui installer, so that it's only installed if the UI is installed:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/ui/installsupport.go#L40